src/run.rs

```diff
@@ -266,22 +267,63 @@ pub fn clean_up(
     tmp_dir: TempDir,
     worktree_branch: Option<&str>,
     subprocesses: &mut [Child],
+    prompt: bool,
 ) -> Result<()> {
     if !delete_file.exists() {
         log::info!(
             "Project directory preserved at: {}",
             tmp_dir.into_path().display()
         );
-    } else if worktree_branch.is_some() {
-        let mut command = std::process::Command::new("git");
-        command
-            .args(["worktree", "remove"])
-            .arg(&tmp_dir.path())
-            .arg("--force");
-        ensure!(
-            command.status().context("Could not start git")?.success(),
-            "cannot remove working tree"
-        );
+    } else {
+        let delete = if prompt {
```

Comment:
I think this whole logic should happen right before you check that the delete file exists or not.

```rs
let mut delete = delete_file.exists();

if prompt {
    // ...
    delete = true//false;
}

if to_delete {
    // delete all the things
} else {
    // preserved and delete delete_file
}
```

When you see redundancy in your code you must rethink the flow from scratch to see if it can be expressed in a way that makes sense without having redundancy.